### PR TITLE
Javacli81

### PR DESCRIPTION
**Pull request by berkeleysquare**
This may not be what was envisioned, but it adds some informative API calls to the CLI and a class (getConfigSummary) to aggregate calls.

Is there a way to access or generate an ASL? That would likely be a better solution to the request.

### CODE REVIEW

File: ds3_java_cli/src/main/java/com/spectralogic/ds3cli/command/GetConfigSummary.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+import com.spectralogic.ds3cli.*;
+import com.spectralogic.ds3cli.exceptions.CommandException;
+import com.spectralogic.ds3cli.models.GetConfigSummaryResult;
+import com.spectralogic.ds3cli.models.Result;
+
+import java.text.SimpleDateFormat;
+import java.util.*;
+
+public class GetConfigSummary extends CliCommand<GetConfigSummaryResult> {
```
Comment (rpmoore) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/command/GetConfigSummary.java]: It would be helpful to have some documentation describing what this command does.

File: ds3_java_cli/src/main/java/com/spectralogic/ds3cli/models/GetCacheStateResult.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+package com.spectralogic.ds3cli.models;
+
+import com.spectralogic.ds3client.models.CacheFilesystemInformation;
+
+import java.util.Iterator;
+import java.util.List;
+
+public class GetCacheStateResult implements Result {
+
+    private List<CacheFilesystemInformation> result;
```
Comment (rpmoore) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/models/GetCacheStateResult.java]: Missing final

File: ds3_java_cli/src/main/java/com/spectralogic/ds3cli/models/GetCapacitySummaryResult.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+import com.spectralogic.ds3client.models.CacheFilesystemInformation;
+import com.spectralogic.ds3client.models.CapacitySummaryContainer;
+
+import java.util.Iterator;
+import java.util.List;
+
+public class GetCapacitySummaryResult implements Result {
+
+    private CapacitySummaryContainer result;
```
Comment (rpmoore) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/models/GetCapacitySummaryResult.java]: Missing final

File: ds3_java_cli/src/main/java/com/spectralogic/ds3cli/models/GetConfigSummaryResult.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+import java.util.Map;
+
+public class GetConfigSummaryResult implements Result {
+
+    // Map for JSON output
+    private final Map<String, Result> result;
+
+    // String for CLI output
+    private String summary;
```
Comment (rpmoore) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/models/GetConfigSummaryResult.java]: missing final

File: ds3_java_cli/src/main/java/com/spectralogic/ds3cli/command/GetConfigSummary.java [hunk 5]
```diff
@@ -0,0 +1,81 @@
+package com.spectralogic.ds3cli.command;
```
Comment (RachelTucker) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/command/GetConfigSummary.java]: missing copyright

File: ds3_java_cli/src/main/java/com/spectralogic/ds3cli/views/cli/GetCacheStateView.java [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+import static com.spectralogic.ds3cli.util.Utils.*;
+
+public class GetCacheStateView extends TableView<GetCacheStateResult> {
+
+    protected List<CacheFilesystemInformation> filesystemList;
+
+    @Override
+    public String render(final GetCacheStateResult obj) {
+        this.filesystemList = obj.getResult();
+        if ((this.filesystemList == null) || (Guard.isNullOrEmpty(this.filesystemList)) ){
```
Comment (rpmoore) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/views/cli/GetCacheStateView.java]: The first null check is not needed when using the guard

File: ds3_java_cli/src/main/java/com/spectralogic/ds3cli/views/json/GetCacheStateView.java [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+import java.util.List;
+
+public class GetCacheStateView implements View<GetCacheStateResult> {
+
+    @Override
+    public String render(final GetCacheStateResult obj) throws JsonProcessingException {
+        final CommonJsonView view = CommonJsonView.newView(CommonJsonView.Status.OK);
+
+        final List<CacheFilesystemInformation> cacheFilesystemInfo = obj.getResult();
+        if( (cacheFilesystemInfo == null) || (cacheFilesystemInfo.size() < 1)){
```
Comment (rpmoore) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/views/json/GetCacheStateView.java]: Use a null guard here instead

File: ds3_java_cli/src/main/java/com/spectralogic/ds3cli/command/CliCommand.java [hunk 8]
```diff
@@ -79,4 +83,16 @@ public String getLongHelp(final String command) {
         return (View<T>) new DefaultView();
     }
 
+    public static String getPlatformInformation() {
+        StringBuilder ret = new StringBuilder();
```
Comment (rpmoore) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/command/CliCommand.java]: Missing final

Comment (rpmoore) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/command/CliCommand.java]: I'm actually surprised that intellij is not complaining about this saying that it can refactored to string concatenation

File: ds3_java_cli/src/main/java/com/spectralogic/ds3cli/views/cli/GetSystemFailureView.java [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+
+public class GetSystemFailureView extends TableView<GetSystemFailureResult> {
+
+    protected List<SystemFailure> failures;
+
+    @Override
+    public String render(final GetSystemFailureResult obj) {
+        final SystemFailureList result = obj.getResult();
+        if( (result == null)
+                || (null == result.getSystemFailures())
```
Comment (rpmoore) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/views/cli/GetSystemFailureView.java]: Since getSystemFailures returns a list, it should use a nullGuard

Comment (berkeleysquare) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/views/cli/GetSystemFailureView.java]: Do I need to first test for null result, or can I be sure that the client would have caught that?

Comment (rpmoore) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/views/cli/GetSystemFailureView.java]: I'd leave the check, but it's probably safe to assume that it won't be.  I would want to prove it first though.
